packages/interflex/src/translate.ts

```diff
@@ -0,0 +1,110 @@
+const PLACEHOLDER = /\{(\w+)\}/g;
+const FRAGMENT = /<(\w+)>(.*?)<\/\1>/g;
+const DEFAULT_DICTIONARY = {
+  localizedName: '',
+  currencySymbol: '',
+  thousandSeparator: '',
+  decimalSeparator: '',
+  decimalPlaces: 0,
+  dateFormat: '',
+  strings: {},
+  monthNames: [],
+  monthNamesShort: [],
+  daysOfWeek: [],
+  daysOfWeekShort: [],
+  startOfWeek: 0,
+};
+let dictionary: LocaleDictionary = {};
+let tLocale: Locale = 'en';
+
+export type Params = {
+  [key: string]:
+    | null
+    | undefined
+    | number
+    | string
+    | boolean
+    | ((s: string) => unknown);
+};
+
+/**
+ * (For testing) to inject data into the dictionary.
+ * @param data
+ */
+export function injectDictionaries(data: LocaleDictionary): void {
+  dictionary = { ...dictionary, ...data };
+}
+
+/**
+ * (For testing) to generate basic dictionary entry with strings content.
+ * @param strings
+ */
+export function generateDefaultDictionary(strings: {
+  [key: string]: string;
+}): LocaleData {
```

Comment:
I'm worried that we're confusing terminology here, between dictionary and localeData.
```suggestion
export function localeDataFromStrings(strings: {
  [key: string]: string;
}): LocaleData {
```